### PR TITLE
fix(avo-1866): adjust permissions

### PR DESCRIPTION
**Pull request by ian-emsens-shd**
https://meemoo.atlassian.net/browse/AVO-1866

https://github.com/viaacode/avo2-client/pull/1333

### CODE REVIEW

File: server/src/modules/data/data.permissions.ts [hunk 1]
```diff
@@ -381,14 +380,19 @@ export const QUERY_PERMISSIONS: {
 		GET_USER_BY_ID: or(PermissionName.EDIT_ANY_USER),
 		GET_USER_TEMP_ACCESS_BY_ID: or(PermissionName.EDIT_ANY_USER),
 		UPDATE_USER_TEMP_ACCESS_BY_ID: or(PermissionName.EDIT_USER_TEMP_ACCESS),
-		GET_USERS: or(PermissionName.EDIT_ANY_USER, PermissionName.EDIT_ANY_COLLECTIONS),
+		GET_USERS: or(
+			PermissionName.EDIT_ANY_USER,
+			PermissionName.EDIT_ANY_COLLECTIONS,
+			PermissionName.VIEW_USERS_IN_SAME_COMPANY
```
Comment (bertyhell) [inline server/src/modules/data/data.permissions.ts:386]: This is a security breach. Users with the permission VIEW_USERS_IN_SAME_COMPANY will still be able to see all the users.

The solution here would be to split the query into 2 queries:
* GET_USERS
* GET_USERS_IN_SAME_COMPANY

The GET_USERS_IN_SAME_COMPANY would look something like this:
```graphql
	query getUsers(
		$offset: Int!
		$limit: Int!
		$orderBy: [users_summary_view_order_by!]!
		$where: users_summary_view_bool_exp!
    $companyId: String!
	) {
		users_summary_view(offset: $offset, limit: $limit, order_by: $orderBy, where: {_and: [{company_id: {_eq: $companyId}}, $where]}) {
			user_id
			full_name
			first_name
			last_name
			mail
			last_access_at
			is_blocked
			blocked_at {
				date: max
			}
			unblocked_at {
				date: max
			}
			profile_id
			stamboek
			acc_created_at
			group_id
			group_name
			company_name
			is_exception
			business_category
			idps {
				idp
			}
			classifications {
				key
			}
			contexts {
				key
			}
			organisations {
				organization_id
				unit_id
				organization {
					ldap_description
				}
			}
			user {
				temp_access {
					until
					from
					current {
						status
					}
				}
			}
		}
		users_summary_view_aggregate(where: {_and: [{company_id: {_eq: $companyId}}, $where]}) {
			aggregate {
				count
			}
		}
	}
```

And in the proxy the permission check for this query would look like this:
```
	GET_USERS_IN_SAME_COMPANY: async (user: Avo.User.User, query: string, variables: any) => {
			if (AuthService.hasPermission(user, PermissionName.EDIT_ANY_USER)) {
				return true;
			}
			if (
				AuthService.hasPermission(user, PermissionName.VIEW_USERS_IN_SAME_COMPANY) &&
				variables.companyId === user.profile.company_id
			) {
				return true;
			}
			return false;
		},
```

**Review by bertyhell — CHANGES_REQUESTED** [review]

File: server/src/modules/data/data.service.ts [hunk 2]
```diff
@@ -192,4 +196,42 @@ export default class DataService {
 			throw new InternalServerError('Failed to fetch collection owner', err);
 		}
 	}
+
+	static transformVariablesByPermission(
```
Comment (bertyhell) [inline server/src/modules/data/data.service.ts:200]: don't create your own permission handling logic, use the standard way of checking permissions in: server/src/modules/data/data.permissions.ts


see remark: https://github.com/viaacode/avo2-proxy/pull/431/files#r870053492

**Review by bertyhell — CHANGES_REQUESTED** [review]

File: server/src/modules/data/data.service.ts (hunk 2 above)
Comment (ian-emsens-shd) [reply server/src/modules/data/data.service.ts:200]: This isn't a permission check, this mutates the query based on which permissions the user does and does not have.

Comment (bertyhell) [reply server/src/modules/data/data.service.ts:200]: It's cleaner to use 2 separate queries with their own permissions.
That way the permissions check happens in a predictable place just like all the other permission checks.

modifying the existing GET_USERS where object based on permissions doesn't follow the existing way of working.

I checked with Bart N. to get his opinion, he also thinks a separate query is more logical.

Comment (ian-emsens-shd) [reply server/src/modules/data/data.service.ts:200]: I disagree but let's go with democracy

**Review by bertyhell — APPROVED** [review]

**Pull request merged**